dspace-api/src/main/java/org/dspace/authorize/AuthorizeServiceImpl.java

```diff
@@ -612,21 +612,10 @@ public void addPolicies(Context c, List<ResourcePolicy> policies, DSpaceObject d
 
         for (ResourcePolicy srp : policies)
         {
-            ResourcePolicy rp = resourcePolicyService.create(c);
-
-            // copy over values
-            rp.setdSpaceObject(dest);
-            rp.setAction(srp.getAction());
-            rp.setEPerson(srp.getEPerson());
-            rp.setGroup(srp.getGroup());
-            rp.setStartDate(srp.getStartDate());
-            rp.setEndDate(srp.getEndDate());
-            rp.setRpName(srp.getRpName());
-            rp.setRpDescription(srp.getRpDescription());
-            rp.setRpType(srp.getRpType());
-
-            // and add policy to list of new policies
-            newPolicies.add(rp);
+            ResourcePolicy policy = resourcePolicyService.create(c);
+            policy = srp.copyToDSpaceObject(policy,dest);
```

Comment:
This kind of beats the purpose of having the "copyToDSpaceObject" method. This code will force hibernate to execute two DB statements: an insert and an update. 

I think you should move the copyToDSpaceObject to the ResourcePolicyService:
* Create a new empty resource policy
* Copy over the values
* Then pass that filled in resource policy to `resourcePolicyDAO.create()`.

You can enable Hibernate DEBUG logging in hibernate.cfg.xml to see the SQL statements it executes.